apps/settings/js/modules/simpin.js

```diff
@@ -0,0 +1,61 @@
+/**
+ * This is a simple model that can help us mimic the status change event
+ * when enabling / disabling simcard.
+ *
+ * @module SimPin
+ */
+define(function(require) {
+  'use strict';
+
+  var Module = require('modules/base/module');
+  var SettingsUtils = require('modules/settings_utils');
+  var EventEmitter = require('modules/base/event_emitter');
+
+  const EVENT_NAME = 'simpin-';
+  const APIS = [
+    'unlockCardLock',
+    'setCardLock',
+    'updateContact',
+    'getCardLock',
+    'getCardLockRetryCount'
+  ];
+
+  var SimPin = Module.create(function() {
+    var supportedEvents = [];
+    APIS.forEach((apiName) => {
+      supportedEvents.push(EVENT_NAME + apiName + '-success');
```

Comment:
The events (names) seem weird to me because they are part of the context of calling a function of this module. Whether the call success or not is only the caller's concern.
I would suggest to emit events reporting sim security state changes (pin enabled/disabled, pin2 enabled/disabled.. etc).